### ISSUE TITLE
`<caption>` element when only `<table>` element is included in `<figure>` element

### ISSUE THREAD

**Issue by SaekiTominaga** [markuplint/markuplint#419]
The `<caption>` element has the following caveats.

> When a table element is the only content in a figure element other than the figcaption, the caption element **should** be omitted in favor of the figcaption.

[4.9.2 The caption element](https://html.spec.whatwg.org/multipage/tables.html#the-caption-element)

```html
<figure>
  <figcaption>figcaption</figcaption>
  <table>
    <caption>caption</caption>
  </table>
</figure>
```

This is currently not warned by Nu Html Checker or markuplint.
(Nu Html Checker has an issue open in August 2021 and appears to be in a status awaiting modification. https://github.com/validator/validator/issues/1208)

Is it possible to get this warning in markuplint?

**Comment by YusukeHirao** [comment]
@SaekiTominaga 

I don't know that the definition is that from the Content Models. I'm still deciding whether I make the new rule or add that to the permitted-content rule. The permitted-content rule checks elements according to normative definitions. I think should make the new rule.

What do you think?

**Comment by SaekiTominaga** [comment]
I think so (new rule).

However, unlike `required-h1`, `id-duplication`..., this rule would apply to minor things.
If adding a new rule has a significant impact on the process, such as making it heavier, then don't have to stick with it.

**Comment by YusukeHirao** [comment]
Thank you. I will create it as a new rule.

**Comment by YusukeHirao** [comment]
@SaekiTominaga 

Improve by https://github.com/markuplint/markuplint/pull/476

So this will resolve when after the release of the next major version.
Please a moment.
